src/core/lib/iomgr/tcp_custom.cc

```diff
@@ -201,12 +172,18 @@ static void endpoint_read(grpc_endpoint* ep, grpc_slice_buffer* read_slices,
   tcp->read_slices = read_slices;
   grpc_slice_buffer_reset_and_unref_internal(read_slices);
   TCP_REF(tcp, "read");
-  if (grpc_slice_allocator_allocate(
-          tcp->slice_allocator, GRPC_TCP_DEFAULT_READ_SLICE_SIZE, 1,
-          grpc_slice_allocator_intent::kReadBuffer, tcp->read_slices,
-          tcp_read_allocation_done, tcp)) {
-    tcp_read_allocation_done(tcp, GRPC_ERROR_NONE);
+  if (tcp->read_slices->length < GRPC_TCP_DEFAULT_READ_SLICE_SIZE) {
+    grpc_slice_buffer_add_indexed(
+        tcp->read_slices, GRPC_SLICE_MALLOC(GRPC_TCP_DEFAULT_READ_SLICE_SIZE));
   }
+  /* Before calling read, we allocate a buffer with exactly one slice
+   * to tcp->read_slices and wait for the callback indicating that the
```

Comment:
Comment is stale, there's no async allocation anymore.

Reply:
fixed